react/features/base/participants/middleware.js

```diff
@@ -229,13 +231,34 @@ StateListenerRegistry.register(
 
                 });
         } else {
-            // We left the conference, raise hand of the local participant must be updated.
-            _raiseHandUpdated(
-                store, conference, undefined, false);
+            // We left the conference, the local participant must be updated.
+            _e2eeUpdated(store, conference, undefined, false);
+            _raiseHandUpdated(store, conference, undefined, false);
         }
     }
 );
 
+/**
+ * Handles a E2EE enabled status update.
+ *
+ * @param {Function} dispatch - The Redux dispatch function.
+ * @param {Object} conference - The conference for which we got an update.
+ * @param {string?} participantId - The ID of the participant from which we got an update. If undefined,
+ * we update the local participant.
+ * @param {boolean} newValue - The new value of the E2EE enabled status.
+ * @returns {void}
+ */
+function _e2eeUpdated({ dispatch, getState }, conference, participantId, newValue) {
+    const e2eeEnabled = newValue === 'true';
+    const pid = participantId || getLocalParticipant(getState()).id;
```

Comment:
hm there's slight risk of triggering update for the local participant in case for whatever reason `participant.getId()` returned `undefined`. How about passing `getLocalParticipant(getState()).id;` as an argument instead like for other participants? or maybe have dedicated method for the local participant which does that

Reply:
well, in fact you're doing it here already, so it makes the flows inconsistent: https://github.com/jitsi/jitsi-meet/pull/6336/commits/f46cedc804762964ef0406fb8d406e301ffc18e0#diff-6b04d343c70445e62f7dad0cf44749d4R27

Reply:
Can do. I had duplicated the logic from raiseHandd, which has been. doing that for a while. I'll adjust both.